specs/Standards/MultiToken/README.md

```diff
@@ -0,0 +1,539 @@
+    ///   approved accounts in case of revert. In this case it may be multiple sets of approvals
+    ///
+    /// Returns true if tokens were successfully transferred to `receiver_id`.
+    fn mt_resolve_transfer(
+        &mut self,
+        sender_id: AccountId,
+        receiver_id: AccountId,
+        token_ids: Vec<TokenId>,
+        amounts: Vec<U128>,
+    ) -> Vec<U128>;
+}
+```
+### Storage Management Trait
+#### Notes
+This is semi necessary for ft token types to be able to refund users for storage of many different token types like gold/silver... this might be slightly out of scope
+```
+pub trait StorageManagement {
+    // if `registration_only=true` MUST refund above the minimum balance if the account didn't exist and
+    //     refund full deposit if the account exists.
+    fn storage_deposit(
+        &mut self,
+        token_ids: Vec<TokenId>,
+        account_id: Option<AccountId>,
+        registration_only: Option<bool>,
+    ) -> StorageBalance;
+
+    /// Withdraw specified amount of available Ⓝ for predecessor account.
+    ///
+    /// This method is safe to call. It MUST NOT remove data.
+    ///
+    /// `amount` is sent as a string representing an unsigned 128-bit integer. If
+    /// omitted, contract MUST refund full `available` balance. If `amount` exceeds
+    /// predecessor account's available balance, contract MUST panic.
+    ///
+    /// If predecessor account not registered, contract MUST panic.
+    ///
+    /// MUST require exactly 1 yoctoNEAR attached balance to prevent restricted
+    /// function-call access-key call (UX wallet security)
+    ///
+    /// Returns the StorageBalance structure showing updated balances.
+    fn storage_withdraw(&mut self, token_ids:Vec<TokenId>, amount: Option<U128>) -> StorageBalance;
+
+    /// Unregisters the predecessor account and returns the storage NEAR deposit back.
+    ///
+    /// If the predecessor account is not registered, the function MUST return `false` without panic.
+    ///
+    /// If `force=true` the function SHOULD ignore account balances (burn them) and close the account.
+    /// Otherwise, MUST panic if caller has a positive registered balance (eg token holdings) or
+    ///     the contract doesn't support force unregistration.
+    /// MUST require exactly 1 yoctoNEAR attached balance to prevent restricted function-call access-key call
+    /// (UX wallet security)
+    /// Returns `true` iff the account was unregistered.
+    /// Returns `false` iff account was not registered before.
+    fn storage_unregister(&mut self, token_ids:Vec<TokenId>, force: Option<bool>) -> Vec<bool>;
+
+    fn storage_balance_bounds(&self, token_id:Vec<TokenId>, account_id: Option<AccountId>) -> StorageBalanceBounds;
+    fn storage_balance_of(&self, token_ids:Vec<TokenId>, account_id: AccountId) -> Option<StorageBalance>;
+}
+```
+
+### Metadata Trait
+```
+pub struct MultiTokenMetadata {
```

Comment:
1. It would be nice if it was clear which of these properties apply to FungibleTokens and which apply to NonFungibleTokens.
2. How does this let us identify if a particular token_id is a NonFungible or Fungible type token?  I would assume that there would be a specific property indicating whether a particular `token_id` represents fungible or non-fungible token?

Reply:
There's no split, not really. We had a long discussion about what was possible and not without leaking implementation details. The multi token standard is about represent all kinds of tokens, so we decided it didn't make sense to have an explicit token type. 

That consumers would have to make their best assumption as to the intent of the developers.  Here's a pointer to the discussion there:

https://github.com/shipsgold/multi-token-standard-impl/blob/main/meetings/2021-08-17-meeting-notes.md#wallet-indexder-3204

Ultimately you are the wallet voice in the room, so you might have more perspective on it than say myself ref,paras, and co. So we want to hear your opinion on this. I believe other wallets have this issue as well such as metamask, generally they allow people to define their collections. 

There are several field that really only make alot sense as fungible such as symbol and decimals, but with this standard there is no hard and fast rule what a FT or NFT or inbetween token is. 

So if you want wallet to represent it as blank... it's up to the consumers to define what they'd like to represent.

Alot of that decision was ideological in not wanting to restrict developers into an explicit understanding of a type. I think the room was flexible bring your thoughts to the next meeting it'll definitely be helpful.